driver-core/src/test/functional/com/mongodb/operation/UserOperationsSpecification.groovy

```diff
@@ -64,85 +63,75 @@ import static com.mongodb.ClusterFixture.isDiscoverableReplicaSet
 import static com.mongodb.ClusterFixture.isSharded
 import static com.mongodb.ClusterFixture.serverVersionAtLeast
 import static com.mongodb.MongoCredential.createCredential
+import static com.mongodb.MongoCredential.createScramSha256Credential
 import static com.mongodb.WriteConcern.ACKNOWLEDGED
 import static java.util.Arrays.asList
 
 class UserOperationsSpecification extends OperationFunctionalSpecification {
     def credential = createCredential('\u53f0\u5317', databaseName, 'Ta\u0301ibe\u030Ci'.toCharArray())
 
     def 'an added user should be found'() {
-        given:
-        new CreateUserOperation(credential, true).execute(getBinding())
-
         when:
-        def found = new UserExistsOperation(databaseName, credential.userName).execute(getBinding())
+        execute(new CreateUserOperation(credential, true), async)
 
         then:
-        found
+        execute(new UserExistsOperation(databaseName, credential.userName), async)
 
         cleanup:
-        new DropUserOperation(databaseName, credential.userName).execute(getBinding())
-    }
-
-    @Category(Async)
-    def 'an added user should be found asynchronously'() {
-        given:
-        executeAsync(new CreateUserOperation(credential, true))
-
-        when:
-        def found = executeAsync(new UserExistsOperation(databaseName, credential.userName))
-
-        then:
-        found
+        execute(new DropUserOperation(databaseName, credential.userName), async)
 
-        cleanup:
-        executeAsync(new DropUserOperation(databaseName, credential.userName))
+        where:
+        async << [true, false]
     }
 
     def 'an added user should authenticate'() {
-        given:
-        new CreateUserOperation(credential, true).execute(getBinding())
-        def cluster = getCluster()
-
         when:
+        execute(new CreateUserOperation(credential, true), async)
+        def cluster = getCluster()
         def server = cluster.selectServer(new WritableServerSelector())
         def connection = server.getConnection()
-        testConnection(connection)
 
         then:
-        connection
+        !testConnection(connection).isEmpty()
 
         cleanup:
         connection?.release()
-        new DropUserOperation(databaseName, credential.userName).execute(getBinding())
+        execute(new DropUserOperation(databaseName, credential.userName), async)
         cluster?.close()
-    }
 
-    @Category(Async)
-    def 'an added user should authenticate asynchronously'() {
-        given:
-        executeAsync(new CreateUserOperation(credential, true))
-        def cluster = getCluster()
+        where:
+        async << [true, false]
+    }
 
+    @IgnoreIf({ !serverVersionAtLeast(3, 7) })
+    def 'should correctly, prep username and be able to authenticate users with unicode in their name'() {
         when:
+        def user = createScramSha256Credential('IX', databaseName, authCredential.getPassword())
```

Comment:
Creating a user requires a Sasl Prepped username - authentication will automatically prep the username.

Reply:
I think this is going to change.  Bernie told me today that it's not feasible to require applications to do their own SASL prepping.

Reply:
I would have thought it would eventually - more of an FYI. It doesn't really matter here as the testcase is still valid.